test/extensions/filters/network/redis_proxy/redis_proxy_integration_test.cc

```diff
@@ -1062,5 +1094,23 @@ TEST_P(RedisProxyWithMirrorsIntegrationTest, EnabledViaRuntimeFraction) {
   redis_client->close();
 }
 
+// This test injects an error fault. The server responds with an error.
+TEST_P(RedisProxyWithFaultInjectionIntegrationTest, ErrorFault) {
+  // TODO: Figure out why there is a memory leak related to ErrorFaultRequest::create
+  std::stringstream fault_response;
+  fault_response << "-" << Extensions::NetworkFilters::Common::Redis::FaultMessages::get().Error
+                 << "\r\n";
+  initialize();
+  simpleProxyResponse(makeBulkStringArray({"get", "foo"}), fault_response.str());
+}
+
+// This test injects a delay fault. The response from the server is unaffected.
+TEST_P(RedisProxyWithFaultInjectionIntegrationTest, DelayFault) {
+  const std::string& set_request = makeBulkStringArray({"set", "write_only:toto", "bar"});
+  const std::string& set_response = ":1\r\n";
+  initialize();
+  simpleRequestAndResponse(set_request, set_response);
```

Comment:
Check the stat after this call that looks for the delay.

Reply:
I'll use the counter for the delay fault, like so `EXPECT_EQ(1, test_server_->counter("redis.redis_stats.command.set.delay_fault")->value());`.